### ISSUE TITLE
bitcoind throw error `Cannot create connection` if there are more than 1024 cocurrent connections 

### ISSUE THREAD

**Issue by greensea** [bitcoin/bitcoin#18158]
Bitcoind throw an error: 
```
Cannot create connection: non-selectable socket created (fd >= FD_SETSIZE ?)
```

**Expected behavior**

Should not display this error.

**To reproduce**
Before start bitcoind, set fd limit to 10240
```
ulimit -n 10240
```
Then start bitcoind with -rpcworkqueue=10240 parameter.
Start to send over 1024 requests to RPC on the same time.

bitcoind version is 0.17.0, OS is Linux.

**Possiable Cause**
I have check the code, if a socket is !IsSelectableSocket(), bitcoind then print this error.
Then I check IsSelectableSocket function in src/compat.h, I found IsSelectableSocket() may return ( fd < FD_SETSIZE) as return value if OS is not Linux. This is strange while I am using Linux.
I think the ``__linux__`` macro is not defined on compiling, but I cant verify that. The binary file from official site.

**Comment by hebasto** [comment]
I think this is not a bug:
https://github.com/bitcoin/bitcoin/blob/94c0287aec9a250f58c4c633dabaf405d82cecd2/src/compat.h#L20-L23

**Comment by gmaxwell** [comment]
@hebasto Why do you think this is not a bug?  Right below, IsSelectableSocket should be return true on linux, and the reporter says they are using the release binaries on linux.  This sounds like it is not correctly using poll anymore, which would be pretty bad.

**Comment by hebasto** [comment]
@greensea 
> bitcoind version is 0.17.0, OS is Linux.
> The binary file from official site.

Which of the following?

```
bitcoin-0.17.0-aarch64-linux-gnu.tar.gz            03-Oct-2018 09:12            27078635
bitcoin-0.17.0-arm-linux-gnueabihf.tar.gz          03-Oct-2018 09:12            24306360
bitcoin-0.17.0-i686-pc-linux-gnu.tar.gz            03-Oct-2018 09:13            30680891
bitcoin-0.17.0-x86_64-linux-gnu.tar.gz             03-Oct-2018 09:15            28262892
```

**Comment by greensea** [comment]
I am using this one:
https://bitcoin.org/bin/bitcoin-core-0.17.0.1/bitcoin-0.17.0.1-x86_64-linux-gnu.tar.gz

I just noticed 0.19 is released, I will upgrade my binary to 0.19 and check if this error still exists. I will report the result later.

**Comment by greensea** [comment]
After upgrade to 0.19.0.1, this error never occurs.
Maybe there is a compile mistake in 0.17.0.1 binary?
Anyway, version 0.19.0.1 works perfectly now.

**Comment by hebasto** [comment]
Is it worth adding to `bitcoind` and `bitcoin-qt` ability to provide a user with compiling flags info (or some important ones of them), e.g., to the `debug.log` file (smth like #17826) ?

**Comment by aureleoules** [comment]
I think this can be closed?

**Comment by MarcoFalke** [comment]
Yeah, looks like this was fixed in 0.19 onward, based on the last comment by the reporter?
